### PR TITLE
[translator/loki] Convert dots to underscores in labels names

### PR DESCRIPTION
**Pull request by mar4uk**
**Description:** <Describe what has changed.>
Loki doesn't support label names containing dots.
Added label names normalization to follow Prometheus label names standard. Dots in label names will be converted to underscores before sending them to Loki

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/14113

### CODE REVIEW

**Review by gouthamve — APPROVED** [review]
LGTM

File: .chloggen/lokiexporter-dots-to-underscores-label-name-normalization.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+note: Added normalization so label name to follow Prometheus label names standard
+
+# One or more tracking issues related to the change
+issues: [14113]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext: |
+  Loki doesn't support label names containing dots. After normalization, all dots in label names are replaced with underscores
```
Comment (kovrus) [inline .chloggen/lokiexporter-dots-to-underscores-label-name-normalization.yaml:17]: Shall we add a sentence telling users that they can drop relabeling from their configuration and that now the `.` -> `_` will happen automatically.

File: exporter/lokiexporter/README.md [hunk 2]
```diff
@@ -78,26 +69,19 @@ attribute as label and the `resource` processor to give a hint to the Loki expor
...
         key: loki.resource.labels
-        value: service_name
+        value: service.name
 ```
 
-Currently, Loki does not support labels with dots. 
-That’s why to add Loki label based on `event.domain` OTLP attribute we need to specify two actions. The first one inserts a new attribute `event_domain` from the OTLP attribute `event.domain`. The second one is a hint for Loki, specifying that the `event_domain` attribute should be placed as a Loki label.
-The same approach is applicable to placing Loki labels from resource attribute `service.name`.
+Currently, Loki does not support label names with dots. 
+That's why lokiexporter normalizes label names to follow Prometheus label names standard before sending requests to Loki.
```
Comment (kovrus) [inline exporter/lokiexporter/README.md:84]: We can point to `pkg/translator/prometheus/README.md#Labels` for more information on labels normalization.

File: .chloggen/lokiexporter-dots-to-underscores-label-name-normalization.yaml (hunk 1 above)
Comment (mar4uk) [reply .chloggen/lokiexporter-dots-to-underscores-label-name-normalization.yaml]: thank you, fixed

File: exporter/lokiexporter/README.md (hunk 2 above)
Comment (mar4uk) [reply exporter/lokiexporter/README.md:84]: added one more sentence with a link to `pkg/translator/prometheus/README.md#Labels`

**Review by kovrus — APPROVED** [review]
lgtm

File: .chloggen/lokiexporter-dots-to-underscores-label-name-normalization.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: enhancement
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: translator/loki
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: Added normalization so label name to follow Prometheus label names standard
```
Comment (evan-bradley) [inline .chloggen/lokiexporter-dots-to-underscores-label-name-normalization.yaml:8]:
```suggestion
note: Normalize label names so they follow the Prometheus label names standard
```
Minor wording suggestion.

**Review by evan-bradley — APPROVED** [review]
Looks good to me.

**Review by codeboten — COMMENTED** [review]
Please resolve conflicts

**Comment by mar4uk** [comment]
@codeboten could you please take another look? conflicts are resolved

**Comment by gouthamve** [comment]
Needs another rebase after https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/18532 unfortunately @mar4uk 😭

**Review by codeboten — APPROVED** [review]

**Pull request merged**